accepted/global-json-updates.md

```diff
@@ -0,0 +1,315 @@
+# Proposal for global.json updates for preview and limited roll-forward
+
+---
+**NOTE:** 
+
+SDK version numbers do not use semantic versioning. [Docs has an article on SDK versioning](https://docs.microsoft.com/en-us/dotnet/core/versions/).
+
+---
+
+`global.json` allows users to pin the version of the SDK they are using. This was added to .NET Core in case we introduced an issue to allow roll-back per project until a fix could be provided. While our guidance remains to use it only for special circumstances, users now use it for numerous reasons.
+
+This document describes changes to `global.json` to provide more flexibility.
+
+## Definitions
+
+The term *SDK patch* is used to describe the last two positions of the SDK major/minor/feature/patch (n in x.y.znn).
+
+The term *feature band* is used to describe the SDK major/minor/feature/patch (x.y.z in x.y.znn).
+
+Given x.y.znn
+
+* x is the major version
+* y is the minor version
+* z is the feature band
+* nn is the patch
+
+## Current behavior
+
+The SDK resolvers (in both the CLI and VS) search from the current location (project in VS) up the directory hierarchy until they encounter a `global.json` file. When the resolver finds one, _it stop looking_ and attempts to resolve the SDK version specified in that file. This happens even if this `global.json` does not specify an SDK version.
+
+When a `global.json` is encountered and it specifies an SDK version, the SDK resolvers attempt to locate that specific SDK patch version:
+
+1. When the resolver finds no `global.json` or finds one that does not specify SDK version, the latest SDK is used, regardless of whether it is preview or stable.
```

Comment:
This doesn't seem to reflect the current behavior. Consider that when building from VS the preview SDK is only used from a preview VS.

Or is this correct in that an RTM VS will pick a preview SDK if there is no global.json but refuse to use a preview SDK if there is a global.json that specifies it?

Reply:
Correct. This reflects the behavior of `dotnet`, but not VS. VS passes in a flag to the same underlying to control when previews are used or not. This flag defaults based on whether VS is a preview and can be overridden by a checkbox in VS. Having a bool in the global.json essentially makes this flag available without the checkbox.

So, `dotnet` can respect it too, and you can set up previews to be used in VS irrespective of checkboxes with this.